airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -2415,10 +2415,13 @@ def var_print(var_name):
             table_id = cmpt[1]
         else:
             raise ValueError(
-                f"{var_print(var_name)} Expect format of (<project.|<project:)<dataset>.<table>, "
+                f"{var_print(var_name)}Expect format of (<project.|<project:)<dataset>.<table>, "
                 f"got {table_input}"
             )
 
+        # Exclude partition from the table name
+        table_id = table_id.split("$")[0]
```

Comment:
@moiseenkov is it possible to have a scenario when `table_id` does not have `$` in a name? I asked, because several months ago we had an issue with the same split code, but for `model_id` and `@` sign. The issue was about incorrect `model_id` because sometimes `model_id` does not have `@` in a name and then this split code assigns an incorrect value.

Reply:
Not having `$` is the most common scenario actually and it is used everywhere in the code. Basically, it is used for referring to a specific partition in the BigQuery table: `table$partition`. Everywhere in the code the table name is expected to be non partitioned as well as in the data lineage integration.

Reply:
@moiseenkov thank you for your explanation. I have checked `split` with values without the `$` sign and see that it works fine and returns the correct value. I was concerned, because for `model_id` we use the `rpartition` function and this function returns an empty string if `model_id` doesn't have a `@` sign. I thought that the `spit` method works the same way :)